[BF16] Add BF16 support to _aminmax and _anminmax_all operators

yusuo opened the pull request:
Summary: Add BF16 support to _aminmax_all and _aminmax operators.

Test Plan:
Added unit test:
https://www.internalfb.com/intern/testinfra/testconsole/testrun/2533274857208373/

Differential Revision: D30073837

anjali411 commented on test/test_reductions.py:
```diff
@@ -667,7 +667,7 @@ def test_amax(self, device, dtype):
     @onlyOnCPUAndCUDA
     @dtypesIfCPU(torch.float, torch.double)
-    @dtypesIfCUDA(torch.half, torch.float)
+    @dtypesIfCUDA(torch.half, torch.float, torch.bfloat16)
```
you should also extend the OpInfo tests for `aminmax` here: https://github.com/pytorch/pytorch/blob/7f1b672b7a1c4ea2b5a71db888047feff22d04af/torch/testing/_internal/common_methods_invocations.py#L6542-L6547 by updating `dtypesIfCUDA` list to include `torch.bfloat16`:

`dtypesIfCUDA=all_types_and(torch.bool, torch.float16, torch.bfloat16),`

yusuo replied:
Done

heitorschueroff commented on aten/src/ATen/native/cuda/ReduceMinMaxKernel.cu:
```diff
@@ -125,7 +125,7 @@ static void aminmax_kernel_impl(
   at::TensorIterator iter = make_reduction("aminmax_cuda", min_result,
     max_result, self, dim, keepdim, self.scalar_type());
-  AT_DISPATCH_ALL_TYPES_AND2(kHalf, kBool, self.scalar_type(), "aminmax_cuda", [&]() {
+  AT_DISPATCH_ALL_TYPES_AND3(kBFloat16, kHalf, kBool, self.scalar_type(), "_aminmax_cuda", [&]() {
```
_aminmax_cuda -> aminmax_cuda

yusuo replied:
Done

heitorschueroff commented on aten/src/ATen/native/cuda/ReduceMinMaxKernel.cu:
```diff
@@ -167,7 +167,7 @@ void aminmax_allreduce_kernel_impl(const Tensor& input, Tensor& min_result, Tens
                              std::vector<int64_t>{}, false, dtype);
   TORCH_CHECK(iter.numel() > 0, "min_max on a tensor with no elements is not defined.");
-  AT_DISPATCH_ALL_TYPES_AND2(kHalf, kBool, dtype, "aminmax_cuda", [&] {
+  AT_DISPATCH_ALL_TYPES_AND3(kBFloat16, kHalf, kBool, dtype, "_aminmax_all_cuda", [&] {
```
_aminmax_all_cuda -> aminmax_all_cuda

yusuo committed:
[BF16] Add BF16 support to _aminmax and _anminmax_all operators (#62767)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/62767

Add BF16 support to _aminmax_all and _aminmax operators.

Test Plan:
Added unit test:
https://www.internalfb.com/intern/testinfra/testconsole/testrun/2533274857208373/

Reviewed By: anjali411

Differential Revision: D30073837

fbshipit-source-id: 0721b8ac7d0e06b9a3686f85dbab9266a26f9178
